lib/rspec/rails/matchers/active_job.rb

```diff
@@ -158,6 +158,9 @@ def queue_adapter
       #       HelloJob.set(wait_until: Date.tomorrow.noon, queue: "low").perform_later(42)
       #     }.to have_enqueued_job.with(42).on_queue("low").at(Date.tomorrow.noon)
       def have_enqueued_job(job = nil)
+        if ::ActiveJob::Base.queue_adapter.class.name != "ActiveJob::QueueAdapters::TestAdapter"
```

Comment:
I think it'd make me and @myronmarston happier if this was
`if ActiveJob::QueueAdapters::TestAdapter === ActiveJob::Base.queue_adapter`